### PR TITLE
Prevent one redirect in `AsynchPeople`

### PR DESCRIPTION
**Pull request by Wadeck**
When clicking on a link from the "People" view, as the links do not include a trailing slash, they are redirected by Stapler to the same url with trailing slash. This could be avoided by just adding the trailing slash when displaying the URL.

Voluntarily not changing `getUrl()` to not break expected behavior (like [MyViewsProperty#L137](https://github.com/jenkinsci/jenkins/blob/e27b310065b3c036b5fc9d123f1d1d99d3058c00/core/src/main/java/hudson/model/MyViewsProperty.java#L137))

_No associated ticket._

### Proposed changelog entries

* Remove one redirect in AsynchPeople

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/View.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/View.java:924]: How about https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#appendIfMissing-java.lang.String-java.lang.CharSequence-java.lang.CharSequence...- ?

Comment (Wadeck) [inline core/src/main/java/hudson/model/View.java:924]:
```suggestion
                        accumulate("url", StringUtils.appendIfMissing(u.getUrl(), "/")).
```

Fortunately the class is already imported in View 🎉 
BUUUUUT, we are including **commons-lang:2.6** ❌ 

IOW, not a good reason to potentially break other stuff just for this quick change.

**Review by MarkEWaite — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
